File: inferences/inference.py
import onnxruntime as ort
import numpy as np
import sys

def infer(model_path):
    print(f"Loading model from {model_path}...")
    session = ort.InferenceSession(model_path)
    
    # Get input name and shape
    input_name = session.get_inputs()[0].name
    input_shape = session.get_inputs()[0].shape
    
    print(f"Expected Input Name: {input_name}")
    print(f"Expected Input Shape: {input_shape}")

    # Prepare a dummy input for testing (Classification expects 3 features)
    # Batch size 1, 3 features.
    X = np.array([[1.0, 2.0, 3.0]], dtype=np.float32)
    print(f"Input Features: {X.tolist()}")

    # Run inference
    result = session.run(None, {input_name: X})
    
    print(f"Output: {result[0].tolist()}")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python infer.py <model.onnx>")
        sys.exit(1)
    infer(sys.argv[1])

File: inferences/inference.ts
import * as ort from 'onnxruntime-node';

async function main() {
    if (process.argv.length < 3) {
        console.error("Usage: npx ts-node inference.ts <path_to_model.onnx>");
        process.exit(1);
    }

    const modelPath = process.argv[2];
    console.log(`Loading model from ${modelPath}...`);

    try {
        // Create an ONNX inference session
        const session = await ort.InferenceSession.create(modelPath);

        // Prepare the input tensor (Dummy Tensor: [1.0, 2.0, 3.0])
        const inputData = Float32Array.from([1.0, 2.0, 3.0]);
        const inputTensor = new ort.Tensor('float32', inputData, [1, 3]);

        // Get the expected input name from the session
        const inputName = session.inputNames[0];
        console.log(`Expected Input Name: ${inputName}`);
        console.log(`Input Features: [1.0, 2.0, 3.0]`);

        // Prepare the feeds
        const feeds: Record<string, ort.Tensor> = {};
        feeds[inputName] = inputTensor;

        // Run the inference session
        const results = await session.run(feeds);

        // Get the output
        const outputName = session.outputNames[0];
        const outputTensor = results[outputName];

        console.log(`Output: [${outputTensor.data[0]}]`);
    } catch (e) {
        console.error(`Failed to infer model: ${e}`);
    }
}

main();
